src/gcp/cloudscheduler.ts

```diff
@@ -12,50 +13,109 @@ export interface Schedule {
     uri: string;
     httpMethod: string;
   };
+  retryConfig?: {
+    retryCount?: number;
+    maxRetryDuration?: string;
+    minBackoffDuration?: string;
+    maxBackoffDuration?: string;
+    maxDoublings?: number;
+  };
 }
-
-export function createJob(schedule: Schedule): Promise<void> {
+/*
+* createJob creates a cloudScheduler job
+* @returns a response from the cloud scheduler api
+* If another job with that name already exists, this will return a 409
+*/
+export function createJob(job: Job): Promise<void> {
   // the replace below removes the portion of the schedule name after the last /
   // ie: projects/my-proj/locations/us-central1/jobs/firebase-schedule-func-us-east1 would become
   // projects/my-proj/locations/us-central1/jobs
-  const strippedName = schedule.name.replace(/\/[^\/]+$/, "");
+  const strippedName = job.name.replace(/\/[^\/]+$/, "");
   return api.request("POST", `/${VERSION}/${strippedName}`, {
     auth: true,
     origin: api.cloudschedulerOrigin,
-    data: Object.assign({ timeZone: "America/Los_Angeles" }, schedule),
+    data: Object.assign({ timeZone: "America/Los_Angeles" }, job),
   });
 }
 
+/*
+* createJob delete a cloudScheduler job with the given name
+* @returns a response from the cloud scheduler api
+* If no job with that name exists, this will return a 404
+*/
 export function deleteJob(name: string): Promise<void> {
   return api.request("DELETE", `/${VERSION}/${name}`, {
     auth: true,
     origin: api.cloudschedulerOrigin,
   });
 }
 
+/*
+* getJob gets a cloudScheduler job with the given name
+* @returns a response from the cloud scheduler api
+* If no job with that name exists, this will return a 404
+*/
 export function getJob(name: string): Promise<any> {
   return api.request("GET", `/${VERSION}/${name}`, {
     auth: true,
     origin: api.cloudschedulerOrigin,
   });
 }
 
-export async function createOrReplaceJob(schedule: Schedule): Promise<void> {
-  const jobName = `${schedule.name.split("/")[5]}`;
+/*
+* updateJob updates a cloudScheduler job
+* @returns a response from the cloud scheduler api
+* If no job with that name exists, this will return a 404
```

Comment:
Returns 404 if job with specified name does not exist. (But move that part to the description).